state/src/account.rs

```diff
@@ -0,0 +1,406 @@
+        }
+    }
+
+    pub fn init_code(&mut self, code: Vec<u8>) {
+        self.code = code.clone();
+        self.code_hash = From::from(&Sha3_256::digest(&code)[..]);
+        self.code_size = code.len();
+        self.code_state = CodeState::Dirty;
+    }
+
+    pub fn read_code<B: DB>(&mut self, db: &mut B) -> Result<Vec<u8>, Error> {
+        if self.code_hash == SHA3_EMPTY {
+            return Ok(vec![]);
+        }
+        if !self.code.is_empty() {
+            return Ok(self.code.clone());
+        }
+        let c = db
+            .get(&self.code_hash)
+            .or(Err(Error::DBError))?
+            .unwrap_or_else(|| vec![]);
+        self.code = c.clone();
+        self.code_size = c.len();
+        self.code_state = CodeState::Clean;
+        Ok(c)
+    }
+
+    pub fn balance(&self) -> U256 {
+        self.balance
+    }
+
+    pub fn nonce(&self) -> U256 {
+        self.nonce
+    }
+
+    pub fn code(&self) -> Option<Vec<u8>> {
+        if self.code.is_empty() {
+            return None;
+        }
+        Some(self.code.clone())
+    }
+
+    pub fn code_hash(&self) -> H256 {
+        self.code_hash
+    }
+
+    pub fn code_size(&self) -> usize {
+        self.code_size
+    }
+
+    pub fn inc_nonce(&mut self) {
+        self.nonce += U256::from(1u8);
+    }
+
+    pub fn add_balance(&mut self, x: U256) {
+        let (a, b) = self.balance.overflowing_add(x);
+        assert_eq!(b, false);
+        self.balance = a;
+    }
+
+    pub fn sub_balance(&mut self, x: U256) {
+        let (a, b) = self.balance.overflowing_sub(x);
+        assert_eq!(b, false);
+        self.balance = a;
+    }
+
+    pub fn set_storage(&mut self, key: H256, value: H256) {
+        self.storage_changes.insert(key, value);
+    }
+
+    pub fn get_storage_at_backend<B: DB>(
+        &mut self,
+        db: &mut B,
+        key: &H256,
+    ) -> Result<Option<H256>, Error> {
+        let trie = PatriciaTrie::from(db, RLPNodeCodec::default(), &self.storage_root.0)
+            .or(Err(Error::DBError))?;
+        if let Ok(a) = trie.get(key) {
+            if let Some(b) = a {
+                return Ok(Some(From::from(&b[..])));
+            }
+        }
+        Ok(None)
+    }
+
+    pub fn get_storage_at_changes(&self, key: &H256) -> Option<H256> {
+        if let Some(value) = self.storage_changes.get(key) {
+            return Some(*value);
+        }
+        None
+    }
+
+    pub fn get_storage<B: DB>(&mut self, db: &mut B, key: &H256) -> Result<Option<H256>, Error> {
+        if let Some(value) = self.get_storage_at_changes(key) {
+            return Ok(Some(value));
+        }
+        if let Some(value) = self.get_storage_at_backend(db, key)? {
+            return Ok(Some(value));
+        }
+        Ok(None)
+    }
+
+    pub fn commit_storage<B: DB>(&mut self, db: &mut B) -> Result<(), Error> {
+        let mut trie = if self.storage_root == SHA3_NULL_RLP {
+            PatriciaTrie::new(db, RLPNodeCodec::default())
```

Comment:
Need to add the `is_empty` method to `cita-trie`?

Reply:
`SHA3_EMPTY` and `SHA3_NULL_RLP` will be used in different modules frequently, I think you should give it directly then `is_empty()`